lib/diff/lcs/ldiff.rb

```diff
@@ -172,11 +172,11 @@ def self.diff?(info_old, info_new, format, output, binary: nil, lines: 0)
     end
 
     last = oldhunk.diff(format, true)
-    last << "\n" if last.respond_to?(:end_with?) && !last.end_with?("\n")
+    last << "\n" if last.respond_to?(:end_with?) && !last.end_with?("\n") && !last.empty?
```

Comment:
`!last.empty?` should probably be first.

Reply:
Actually...

1️⃣ NIT: the `! empty?` case is interesting only if the `! end_with?("\n")`, having it first would do 2 checks while in many (most?) cases there will be a non-empty last diff.

2️⃣ I had trouble to understand why the `respond_to?(!:end_with?)` was necessary, and it is in the case where you request `ed` formatted diff ... instead of returning the diff, this is the `Hunk` which is returned, to be processed "in reversed order" L179.
So if I put `!last.empty?` first I risk to get `no method .empty? for Hunk class`, I could guard it with a `.respond_to?(:empty?)` but it is a bit heavy.

➡️ I can do `if !last.is_a?(Hunk) && !last.empty? && last.end_with?("\n")`. Would that be OK?

Reply:
In the end I did `unless last.is?(Hunk) || last.empty? || last.end_with?("\n")`, let me know if this is OK